crates/gpui/src/app.rs

```diff
@@ -824,8 +828,16 @@ impl App {
     }
 
     /// Restarts the application.
-    pub fn restart(&self, binary_path: Option<PathBuf>) {
-        self.platform.restart(binary_path)
+    pub fn restart(&mut self) {
+        self.restart_observers
+            .clone()
+            .retain(&(), |observer| observer(self));
+        self.platform.restart(self.restart_path.clone())
```

Comment:
```suggestion
        self.platform.restart(self.restart_path.take())
```
Doing a direct `take()` here should avoid an unnecessary string copy. After this function call, we’re never going to use this variable again anyway, Zed’s already restarted, after all.